src/main/java/com/palantir/docker/compose/connection/ContainerNames.java

```diff
@@ -16,98 +16,26 @@
 
 import java.util.Arrays;
-import java.util.Iterator;
 import java.util.List;
-import java.util.Objects;
-import java.util.function.Function;
-import java.util.stream.Stream;
 
-public class ContainerNames implements Iterable<String> {
+public class ContainerNames {
 
-    private final List<String> containerNames;
+    private ContainerNames() {}
 
-    public ContainerNames(String singleContainerName) {
-        this(singletonList(singleContainerName));
-    }
-
-    public ContainerNames(List<String> containerNames) {
-        this.containerNames = containerNames;
-    }
-
-    public static ContainerNames parseFromDockerComposePs(String psOutput) {
+    public static List<ContainerName> parseFromDockerComposePs(String psOutput) {
         String[] splitOnSeparator = psOutput.split("-+\n");
         if (splitOnSeparator.length < 2) {
-            return new ContainerNames(emptyList());
+            return emptyList();
         }
-        return new ContainerNames(getContainerNamesAtStartOfLines(splitOnSeparator[1]));
-    }
-
-    private static List<String> getContainerNamesAtStartOfLines(String psContainerOutput) {
-        return Arrays.stream(psContainerOutput.split("\n"))
-                     .map(String::trim)
-                     .filter(line -> !line.isEmpty())
-                     .map(line -> line.split(" "))
-                     .map(psColumns -> psColumns[0])
-                     .map(withoutDirectory().andThen(withoutScaleNumber()))
-                     .collect(toList());
-    }
-
-    public static Function<String, String> withoutDirectory() {
-        return fullname -> Arrays.stream(fullname.split("_"))
-                .skip(1)
-                .collect(joining("_"));
-    }
-
-    public static Function<String, String> withoutScaleNumber() {
-        return fullname -> {
-            final String[] components = fullname.split("_");
-            return Arrays.stream(components)
-                    .limit(components.length - 1)
-                    .collect(joining("_"));
-        };
-    }
-
-    @Override
-    public Iterator<String> iterator() {
-        return containerNames.iterator();
-    }
-
-    public Stream<String> stream() {
-        return containerNames.stream();
-    }
-
-    public int size() {
-        return containerNames.size();
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(containerNames);
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (obj == null) {
-            return false;
-        }
-        if (getClass() != obj.getClass()) {
-            return false;
-        }
-        ContainerNames other = (ContainerNames) obj;
-        return Objects.equals(containerNames, other.containerNames);
-    }
-
-    @Override
-    public String toString() {
-        return "ContainerNames [containerNames=" + containerNames + "]";
+        String psBody = splitOnSeparator[1];
```

Comment:
Can you put this in a method? I don't really know what's going on even after reading it.